Jenkinsfile

```diff
@@ -25,19 +15,99 @@ pipeline {
+                        sleep 10
+                        sh label: 'Show all running containers', script: 'docker ps'
+                    }
+                }
+                stage('Run tests') {
+                    steps {
+                        sh label: 'Running tests', script: 'docker-compose run nhais-tests'
+                    }
+                    post {
+                        always {
+                            sh label: 'Copy test reports to folder', script: 'docker cp "$(docker ps -lq)":/usr/src/app/nhais/test-reports .'
+                            junit '**/test-reports/*.xml'
+                            sh label: 'Copy test coverage to folder', script: 'docker cp "$(docker ps -lq)":/usr/src/app/nhais/coverage.xml ./coverage.xml'
+                            cobertura coberturaReportFile: '**/coverage.xml'
+                        }
+                    }
+                }
+                stage('Push image') {
+                    when {
+                        expression { currentBuild.resultIsBetterOrEqualTo('SUCCESS') }
+                    }
+                    steps {
+                        script {
+                            sh label: 'Pushing nhais image', script: "packer build -color=false pipeline/packer/nhais.json"
+                        }
+                    }
+                }
+            }
+            post {
+                always {
+                    sh label: 'Create logs directory', script: 'mkdir logs'
+                    sh label: 'Copy nhais container logs', script: 'docker-compose logs nhais > logs/nhais.log'
+                    sh label: 'Copy dynamo container logs', script: 'docker-compose logs dynamodb > logs/outbound.log'
+                    sh label: 'Copy rabbitmq logs', script: 'docker-compose logs rabbitmq > logs/inbound.log'
+                    sh label: 'Copy nhais-tests logs', script: 'docker-compose logs nhais-tests > logs/nhais-tests.log'
+                    archiveArtifacts artifacts: 'logs/*.log', fingerprint: true
+                    sh label: 'Stopping containers', script: 'docker-compose down -v'
+                }
+            }
+        }
+        stage('Deploy and Integration Test') {
             when {
-              expression { runSonarQube }
-            } 
+                expression { currentBuild.resultIsBetterOrEqualTo('SUCCESS') }
+            }
+            stages {
+                stage('Deploy using Terraform') {
+                    steps {
+                        echo 'TODO deploy NHAIS using terraform'
+                    }
+                }
+                stage('Run integration tests') {
+                    steps {
+                        echo 'TODO run integration tests'
+                        echo 'TODO archive test results'
+                    }
+                 }
+            }
+
+        }
+        stage('Run SonarQube analysis') {
             steps {
                 dir('.') {
```

Comment:
Why changing direcitory to '.' (current directory)?

Reply:
Good spot, I have removed this.